File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int n=nums.length;
        int minLength=n+2;

        int windowstart=0;
        int sum=0;

        for(int i=0;i<n;i++){
            sum+=nums[i];

            // if(sum>=target){
                while(target<=sum){
                    minLength=Math.min(minLength,i-windowstart+1);
                    sum-=nums[windowstart];
                    windowstart++;
                }
            // }
        }

        if(minLength==n+2) return 0;
        return minLength;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
var minSubArrayLen = function(target, nums) {
    let l=0;
    let r=0;
    let minLength=Infinity;
    let sum=0;

    while(r<nums.length){
        sum+=nums[r];

        while(sum>=target && l<=r){
            minLength=Math.min(minLength,r-l+1);
            sum-=nums[l];
            l++;
        }
        r++;
    }

    if(minLength == Infinity) return 0;
    return minLength;
};